pkg/controller/whisker/whisker_controller.go

```diff
@@ -0,0 +1,279 @@
+		// ensures that we don't surprise anyone by changing defaults in a future version of the operator.
+		if err := r.cli.Patch(ctx, managementClusterConnection, preDefaultPatchFrom); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, err.Error(), err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		log.V(2).Info("Loaded ManagementClusterConnection config", "config", managementClusterConnection)
+
+		if managementClusterConnection.Spec.TLS.CA == operatorv1.CATypePublic {
+			// If we need to trust a public CA, then we want Guardian to mount all the system certificates.
+			trustedCertBundle, err = certificateManager.CreateTrustedBundleWithSystemRootCertificates()
+			if err != nil {
+				r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create tigera-ca-bundle configmap", err, reqLogger)
+				return reconcile.Result{}, err
+			}
+		} else {
+			trustedCertBundle = certificateManager.CreateTrustedBundle()
+		}
+	}
+
+	if trustedCertBundle == nil {
+		trustedCertBundle = certificateManager.CreateTrustedBundle()
+	}
+
+	secretsToTrust := []string{render.ProjectCalicoAPIServerTLSSecretName(installation.Variant)}
+	for _, secretName := range secretsToTrust {
+		secret, err := certificateManager.GetCertificate(r.cli, secretName, common.OperatorNamespace())
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, fmt.Sprintf("Failed to retrieve %s", secretName), err, reqLogger)
+			return reconcile.Result{}, err
+		} else if secret == nil {
+			reqLogger.Info(fmt.Sprintf("Waiting for secret '%s' to become available", secretName))
+			r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Waiting for secret '%s' to become available", secretName), nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		trustedCertBundle.AddCertificates(secret)
+	}
+
+	ch := utils.NewComponentHandler(log, r.cli, r.scheme, whiskerCR)
+	cfg := &whisker.Configuration{
+		PullSecrets:                 pullSecrets,
+		OpenShift:                   r.provider.IsOpenShift(),
+		Installation:                installation,
+		TunnelSecret:                tunnelSecret,
+		TrustedCertBundle:           trustedCertBundle,
+		LinseedPublicCASecret:       linseedCASecret,
+		ManagementClusterConnection: managementClusterConnection,
+	}
+
+	components := []render.Component{whisker.Whisker(cfg)}
+	if err = imageset.ApplyImageSet(ctx, r.cli, variant, components...); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	for _, component := range components {
+		if err := ch.CreateOrUpdateOrDelete(ctx, component, r.status); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+			return result, err
+		}
+	}
+
+	r.status.ClearDegraded()
+
+	return result, nil
+}
+
+func newNSObjectKey(name, namespace string) client.ObjectKey {
```

Comment:
Do we really need this function? Seems more or less just as verbose as inlineing `types.NamespacedName{Name: name, Namespace: ns}` but a lot less readable.

Reply:
removed